pkg/cmd/util/clientcmd/factory_object_mapping.go

```diff
@@ -0,0 +1,460 @@
+			client.UsersInterface(oc),
+			client.GroupsInterface(oc),
+			client.ClusterRoleBindingsInterface(oc),
+			client.RoleBindingsNamespacer(oc),
+			client.OAuthClientAuthorizationsInterface(oc),
+			kc.Core(),
+		), nil
+	case userapi.Kind("Group"):
+		oc, kc, err := f.Clients()
+		if err != nil {
+			return nil, err
+		}
+		return authenticationreaper.NewGroupReaper(
+			client.GroupsInterface(oc),
+			client.ClusterRoleBindingsInterface(oc),
+			client.RoleBindingsNamespacer(oc),
+			kc.Core(),
+		), nil
+	case buildapi.Kind("BuildConfig"):
+		oc, _, err := f.Clients()
+		if err != nil {
+			return nil, err
+		}
+		return buildcmd.NewBuildConfigReaper(oc), nil
+	}
+	return f.kubeObjetMappingFactory.Reaper(mapping)
+}
+
+func (f *ring1Factory) HistoryViewer(mapping *meta.RESTMapping) (kubectl.HistoryViewer, error) {
+	switch mapping.GroupVersionKind.GroupKind() {
+	case deployapi.Kind("DeploymentConfig"):
+		oc, kc, err := f.Clients()
+		if err != nil {
+			return nil, err
+		}
+		return deploycmd.NewDeploymentConfigHistoryViewer(oc, kc), nil
+	}
+	return f.kubeObjetMappingFactory.HistoryViewer(mapping)
+}
+
+func (f *ring1Factory) Rollbacker(mapping *meta.RESTMapping) (kubectl.Rollbacker, error) {
+	switch mapping.GroupVersionKind.GroupKind() {
+	case deployapi.Kind("DeploymentConfig"):
+		oc, _, err := f.Clients()
+		if err != nil {
+			return nil, err
+		}
+		return deploycmd.NewDeploymentConfigRollbacker(oc), nil
+	}
+	return f.kubeObjetMappingFactory.Rollbacker(mapping)
+}
+
+func (f *ring1Factory) StatusViewer(mapping *meta.RESTMapping) (kubectl.StatusViewer, error) {
+	oc, _, err := f.Clients()
+	if err != nil {
+		return nil, err
+	}
+
+	switch mapping.GroupVersionKind.GroupKind() {
+	case deployapi.Kind("DeploymentConfig"):
+		return deploycmd.NewDeploymentConfigStatusViewer(oc), nil
+	}
+	return f.kubeObjetMappingFactory.StatusViewer(mapping)
+}
+
+func (f *ring1Factory) AttachablePodForObject(object runtime.Object) (*kapi.Pod, error) {
+	switch t := object.(type) {
+	case *deployapi.DeploymentConfig:
+		_, kc, err := f.Clients()
+		if err != nil {
+			return nil, err
+		}
+		selector := labels.SelectorFromSet(t.Spec.Selector)
+		f := func(pods []*kapi.Pod) sort.Interface { return sort.Reverse(controller.ActivePods(pods)) }
+		pod, _, err := kcmdutil.GetFirstPod(kc, t.Namespace, selector, 1*time.Minute, f)
+		return pod, err
+	default:
+		return f.kubeObjetMappingFactory.AttachablePodForObject(object)
+	}
+}
+
+// PrinterForMapping returns a printer suitable for displaying the provided resource type.
+// Requires that printer flags have been added to cmd (see AddPrinterFlags).
+func (f *ring1Factory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error) {
+	// TODO FIX ME. COPIED FROM KUBE AS PART OF THE COPY/PASTE FOR
+	// PrinterForMapping
+	if latest.OriginKind(mapping.GroupVersionKind) {
+		printer, ok, err := kcmdutil.PrinterForCommand(cmd)
```

Comment:
@deads2k is there a smaller, cleaner way to do this so we don't have to copy & paste?